cpp/src/lance/format/schema.cc

```diff
@@ -499,6 +499,19 @@ ::arrow::Result<std::shared_ptr<Schema>> Schema::Project(
   return projection;
 }
 
+::arrow::Result<std::shared_ptr<Schema>> Schema::Project(
+    const ::arrow::compute::Expression& expr) const {
```

Comment:
Does input need to be bound?

Reply:
It does not. This just read the schema and run projections. 
Bound expr is only needed to feed into `arrow::compute`